opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
             one file per iteration, and the name of the files should end
             with the iteration number, followed by '.h5' (e.g. data0005000.h5)
         """
-        OpenPMDTimeSeries.__init__( path_to_dir )
+        super(LpaDiagnostics, self).__init__( path_to_dir )
+
+    def get_rms_gamma( self, t=None, iteration=None, species=None,
+                       select=None ):
+        """
+        Calculate the rms gamma and standard deviation according to the
+        particle weights
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A tuple of floats with:
+        - mean weighted gamma
+        - weighted standard deviation of gamma
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        ux, uy, uz, w = self.get_particle(
+                         var_list=['ux', 'uy', 'uz', 'w'],
+                         species=species, t=t, iteration=iteration )
+        # Calculate Lorentz factor for all particles
+        gamma = np.sqrt(1 + ux ** 2 + uy ** 2 + uz ** 2)
+        # Calculate weighted mean and average
+        try:
+            # Calculate mean_gamma for selected particles
+            mean_gamma = np.sqrt(np.average(gamma ** 2, weights=w))
```

Comment:
I was wondering if there is a particular reason to calculate the RMS of gamma here, instead of the actual mean (i.e. `np.average(gamma, weights=w)`) - which would maybe be more standard.